PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,346 @@
+//
+//  EntitlementInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 6/25/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+/**
+ Enum of supported stores
+ */
+@objc(RCStore) public enum Store: Int {
+    /// For entitlements granted via Apple App Store.
+    @objc(RCAppStore) case appStore = 0
+    /// For entitlements granted via Apple Mac App Store.
+    @objc(RCMacAppStore) case macAppStore = 1
+    /// For entitlements granted via Google Play Store.
+    @objc(RCPlayStore) case playStore = 2
+    /// For entitlements granted via Stripe.
+    @objc(RCStripe) case stripe = 3
+    /// For entitlements granted via a promo in RevenueCat.
+    @objc(RCPromotional) case promotional = 4
+    /// For entitlements granted via an unknown store.
+    @objc(RCUnknownStore) case unknownStore = 5
+}
+
+/**
+ Enum of supported period types for an entitlement.
+ */
+@objc(RCPeriodType) public enum PeriodType: Int {
+    /// If the entitlement is not under an introductory or trial period.
+    @objc(RCNormal) case normal = 0
+    /// If the entitlement is under a introductory price period.
+    @objc(RCIntro) case intro = 1
+    /// If the entitlement is under a trial period.
+    @objc(RCTrial) case trial = 2
+}
+
+@objcMembers @objc(RCEntitlementInfo) public class EntitlementInfo: NSObject {
```

Comment:
nice, didn't know about `@objcMembers` but it seems super useful

Reply:
I assume it only applies to the current declaration? i.e.: it wouldn't apply for extensions unless we also mark those as `@objcMembers`?

Reply:
also, perhaps we should strive to only add in this declaration the stuff we actually want to expose to obj-c, and move the class funcs to an extension, since they don't even need to be obj-c compatible

Reply:
@aboedo what wouldn't need to be exposed to obj-c? anything we wouldn't want exposed to users? in that case, could we not just make it private?

Reply:
like with everything, we should only expose what we absolutely need to. 
In this case, we need to expose whatever was a part of the original API in objective-c. 
and yes, making it either `private` or `internal` would do the trick, I left a comment related to that in one of the methods that might not need to be exposed

Reply:
`@objcMembers`:
Apply this attribute to a class declaration, to implicitly apply the objc attribute to all Objective-C compatible members of the class, its extensions, its subclasses, and all of the extensions of its subclasses.

Yeah, so maybe I'll do it member-wise, instead 😄